R/capture-requests.R

```diff
@@ -260,24 +260,46 @@ dbGetInfoConTrace <- quote({
   }
 })
 
-dbGetInfoResultTrace <- quote({
-  thing <- returnValue()
-  # TODO: would this be better if we traced the methods individually?
-  if (inherits(dbObj, c("MariaDBResult", "PqResult", "SQLiteResult"))) {
-    hash <- hash(dbObj@sql)
-  } else if (inherits(dbObj, "OdbcResult")) {
-    hash <- hash(dbObj@statement)
+#' Extract a hash from a (result) object
+#'
+#' This function should generally not be used, but must be exported for the
+#' query recording function to work properly
+#'
+#' @param obj the DBI result object to hash
+#'
+#' @return hash for the object
+#' @export
+#' @keywords internal
+extract_hash_from_object <- function(obj) {
+  # TODO: would this be better if we traced the methods using signature?
+  if (inherits(obj, "PostgreSQLResult")) {
+    result_info <- RPostgreSQL::postgresqlResultInfo(obj)
+    hash <- hash(result_info$statement)
+  } else if (inherits(obj, c("MariaDBResult", "PqResult", "SQLiteResult"))) {
+    hash <- hash(obj@sql)
+  } else if (inherits(obj, "OdbcResult")) {
+    hash <- hash(obj@statement)
   } else {
-    # TODO: some default?
+    # Stringify the result to get a hash is better than nothing
+    # If no hash is found, there is an error:
+    # Error in paste0(type, "-", hash, ".R") :
+    #   cannot coerce type 'closure' to vector of type 'character'
```

Comment:
```suggestion
```

Thanks for the detailed comment — we'll have this info in the PR for posterity so I think just the first line is needed here now. (But I very veyr much appreciate the intent to be super explicit!)